sdk/python/feast/infra/utils/postgres/connection_utils.py

```diff
@@ -63,16 +72,14 @@ def df_to_postgres_table(
     """
     Create a table for the data frame, insert all the values, and return the table schema
     """
+    nr_columns = df.shape[1]
+    placeholders = ", ".join(["%s"] * nr_columns)
+    query = f"INSERT INTO {table_name} VALUES ({placeholders})"
+    values = df.replace({np.NaN: None}).to_numpy().tolist()
+
     with _get_conn(config) as conn, conn.cursor() as cur:
         cur.execute(_df_to_create_table_sql(df, table_name))
-        psycopg2.extras.execute_values(
-            cur,
-            f"""
-            INSERT INTO {table_name}
-            VALUES %s
-            """,
-            df.replace({np.NaN: None}).to_numpy(),
-        )
+        cur.executemany(query, values)
```

Comment:
Moved the parsing of variables further to the top of the function. 

1. Again, we need to replace `execute_values` by `executemany`. 
2. Again, we need to explicitly set the number of placeholders. Since this function should be able to handle a dynamic amount of columns, we use the `placeholders` variable